### PR TITLE
W2 - tungnguyen - Create RecyclerView 

### PR DESCRIPTION
**Pull request by at-tungnguyen2**
#### Related issues (Optional)
Task #21

#### Description (Required)
- [x] Create screen
- [x] Defines models class
- [x] Bind Data
- [x] Handle button and set color count like

### CODE REVIEW

File: app/src/at-tungnguyen/java/vn/asiantech/internship/models/PersonAdapter.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
+package vn.asiantech.internship.models;
+
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
```
Comment (at-dev-support-01) [inline app/src/at-tungnguyen/java/vn/asiantech/internship/models/PersonAdapter.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.widget.AdapterView'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/at-tungnguyen/java/vn/asiantech/internship/models/PersonAdapter.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import android.widget.ImageView;
+import android.widget.LinearLayout;
```
Comment (at-dev-support-01) [inline app/src/at-tungnguyen/java/vn/asiantech/internship/models/PersonAdapter.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.widget.LinearLayout'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

**Comment by at-tungnguyen2** [comment]
@at-huongnguyen  Please check for me, thank.

File: app/src/at-tungnguyen/java/vn/asiantech/internship/MainActivity.java [hunk 3]
```diff
@@ -3,13 +3,53 @@
...
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        initAdapter();
+        initData();
+        initViews();
+    }
+
+    private void initAdapter() {
+        mAdapter = new StatusAdapter(mStatusList, this) {};
```
Comment (at-huongnguyen) [inline app/src/at-tungnguyen/java/vn/asiantech/internship/MainActivity.java]: Remove `{}`

File: app/src/at-tungnguyen/java/vn/asiantech/internship/MainActivity.java [hunk 4]
```diff
@@ -3,13 +3,53 @@
...
+    private void initData() {
+        mStatusList.add(new Status("Tung", "Messi messi", 0));
+        mStatusList.add(new Status("Tung", "messi messi messi", 0));
+        mStatusList.add(new Status("Tung", "messi messi messi messi messi messi", 0));
+    }
+
+    private void initViews() {
+        RecyclerView mRecyclerViewStatus = findViewById(R.id.recyclerViewPerson);
+        mRecyclerViewStatus.setLayoutManager(new LinearLayoutManager(this));
+        mRecyclerViewStatus.setAdapter(mAdapter);
```
Comment (at-huongnguyen) [inline app/src/at-tungnguyen/java/vn/asiantech/internship/MainActivity.java]: Line 40, 41 move to initAdapter method

File: app/src/at-tungnguyen/java/vn/asiantech/internship/MainActivity.java [hunk 5]
```diff
@@ -3,13 +3,53 @@
...
+import android.support.v7.widget.RecyclerView;
 
-public class MainActivity extends Activity {
+import java.util.ArrayList;
+import java.util.List;
+
+import vn.asiantech.internship.models.Status;
+import vn.asiantech.internship.models.StatusAdapter;
+
+public class MainActivity extends Activity implements StatusAdapter.OnItemClickListener {
```
Comment (at-huongnguyen) [inline app/src/at-tungnguyen/java/vn/asiantech/internship/MainActivity.java]: Add line space

File: app/src/at-tungnguyen/java/vn/asiantech/internship/models/StatusAdapter.java [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    private OnItemClickListener mOnItemClickListener;
+
+    protected StatusAdapter(List<Status> statusList, OnItemClickListener onItemClickListener) {
+        mStatusList = statusList;
+        mOnItemClickListener = onItemClickListener;
+    }
+
+    @Override
+    public StatusAdapter.StatusViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_person, parent, false);
```
Comment (at-huongnguyen) [inline app/src/at-tungnguyen/java/vn/asiantech/internship/models/StatusAdapter.java]: Should rename to: item_status

File: app/src/at-tungnguyen/java/vn/asiantech/internship/models/StatusAdapter.java [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+            }
+        }
+
+        @Override
+        public void onClick(View view) {
+            switch (view.getId()) {
+                case R.id.imgLike:
+                    if (mOnItemClickListener != null) {
+                        mOnItemClickListener.onLikeClick(getAdapterPosition());
+                        break;
```
Comment (at-huongnguyen) [inline app/src/at-tungnguyen/java/vn/asiantech/internship/models/StatusAdapter.java]: Move break go out if{}

File: app/src/at-tungnguyen/java/vn/asiantech/internship/models/StatusAdapter.java [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+                    }
+                case R.id.imgDislike:
+                    if (mOnItemClickListener != null) {
+                        mOnItemClickListener.onDisLikeClick(getAdapterPosition());
+                        break;
```
Comment (at-huongnguyen) [inline app/src/at-tungnguyen/java/vn/asiantech/internship/models/StatusAdapter.java]: Move break go out if{}

File: app/src/at-tungnguyen/res/drawable/selector_dislike_sub.xml [hunk 9]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
+    <item android:drawable="@drawable/bg_urbi_reverseb" android:state_pressed="true"></item>
+    <item android:drawable="@drawable/bg_urbi_reverseb" android:state_selected="true"/>
+    <item android:drawable="@drawable/ic_dislike"></item>
```
Comment (at-huongnguyen) [inline app/src/at-tungnguyen/res/drawable/selector_dislike_sub.xml]: Use ..../>

File: app/src/at-tungnguyen/res/drawable/selector_like_plus.xml [hunk 10]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
+    <item android:drawable="@drawable/ic_click_like" android:state_pressed="true"></item>
+    <item android:drawable="@drawable/ic_click_like" android:state_selected="true"></item>
+    <item android:drawable="@drawable/ic_like_ok"></item>
```
Comment (at-huongnguyen) [inline app/src/at-tungnguyen/res/drawable/selector_like_plus.xml]: Use ..../>

File: app/src/at-tungnguyen/res/layout/activity_main.xml [hunk 11]
```diff
@@ -1,9 +1,13 @@
 <?xml version="1.0" encoding="utf-8"?>
-<RelativeLayout
-    xmlns:android="http://schemas.android.com/apk/res/android"
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (at-huongnguyen) [inline app/src/at-tungnguyen/res/layout/activity_main.xml]: Remove ViewGroup

File: app/src/at-tungnguyen/res/layout/item_person.xml [hunk 12]
```diff
@@ -0,0 +1,79 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (at-huongnguyen) [inline app/src/at-tungnguyen/res/layout/item_person.xml]: Remove ViewGroup

File: app/src/at-tungnguyen/res/layout/item_person.xml [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+    android:orientation="horizontal"
+    android:padding="5dp">
+
+    <android.support.v7.widget.CardView
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content">
+
+        <LinearLayout
+            android:layout_width="match_parent"
+            android:layout_height="120dp"
```
Comment (at-huongnguyen) [inline app/src/at-tungnguyen/res/layout/item_person.xml]: Use wrap_content

File: app/src/at-tungnguyen/res/layout/item_person.xml [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+            android:baselineAligned="false"
+            android:weightSum="10">
```
Comment (at-huongnguyen) [inline app/src/at-tungnguyen/res/layout/item_person.xml]: Remove weightSum

File: app/src/at-tungnguyen/res/layout/item_person.xml [hunk 15]
```diff
@@ -0,0 +1,79 @@
...
+
+            <LinearLayout
```
Comment (at-huongnguyen) [inline app/src/at-tungnguyen/res/layout/item_person.xml]: Remove ViewGroup

File: app/src/at-tungnguyen/res/layout/item_person.xml [hunk 16]
```diff
@@ -0,0 +1,79 @@
...
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_marginTop="10dp"
+                    android:textColor="@android:color/black"
+                    android:textSize="16dp"
+                    android:textStyle="bold"
+                    tools:text="Thanh Tung" />
+            </LinearLayout>
+
+            <LinearLayout
```
Comment (at-huongnguyen) [inline app/src/at-tungnguyen/res/layout/item_person.xml]: Remove ViewGroup

File: app/src/at-tungnguyen/res/layout/item_person.xml [hunk 17]
```diff
@@ -0,0 +1,75 @@
...
+    app:cardCornerRadius="3dp">
+
+    <LinearLayout
+        android:layout_marginTop="10dp"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:baselineAligned="false"
+        android:weightSum="10">
+
+        <LinearLayout
```
Comment (at-dev-support-01) [inline app/src/at-tungnguyen/res/layout/item_person.xml]: WARNING: [UselessParent] This `LinearLayout` layout or its `LinearLayout` parent is useless
 A layout with children that has no siblings, is not a scrollview or a root layout, and does not have a background, can be removed and have its children moved directly into the parent for a flatter and more efficient layout hierarchy.

File: app/src/at-tungnguyen/java/vn/asiantech/internship/MainActivity.java [hunk 18]
```diff
@@ -3,13 +3,56 @@
...
+        initData();
+        initViews();
+    }
+
+    private void initAdapter() {
+        mAdapter = new StatusAdapter(mStatusList, this) {};
     }
 
+    private void initData() {
+        mStatusList.add(new Status("Tung", "Messi messi", 0));
```
Comment (at-dev-support-01) [inline app/src/at-tungnguyen/java/vn/asiantech/internship/MainActivity.java]: WARNING: [AvoidDuplicateLiterals] The String literal "Tung" appears 6 times in this file; the first occurrence is on line 33
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/strings.html#AvoidDuplicateLiterals

**Comment by at-tungnguyen2** [comment]
@at-huongnguyen  Please check for me, thank you.

**Comment by at-huongnguyen** [comment]
@at-tungnguyen2 
Please resolve conflicts

File: app/src/at-tungnguyen/java/vn/asiantech/internship/CreateLogin/LoginFragment.java [hunk 19]
```diff
@@ -1,4 +1,4 @@
-package vn.asiantech.internship;
+package vn.asiantech.internship.CreateLogin;
```
Comment (at-dev-support-01) [inline app/src/at-tungnguyen/java/vn/asiantech/internship/CreateLogin/LoginFragment.java]: WARNING: [PackageCase] Package name contains upper case characters
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#PackageCase

File: app/src/at-tungnguyen/java/vn/asiantech/internship/CreateLogin/MainActivity.java [hunk 20]
```diff
@@ -1,5 +1,5 @@
 
-package vn.asiantech.internship;
+package vn.asiantech.internship.CreateLogin;
```
Comment (at-dev-support-01) [inline app/src/at-tungnguyen/java/vn/asiantech/internship/CreateLogin/MainActivity.java]: WARNING: [PackageCase] Package name contains upper case characters
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#PackageCase

File: app/src/at-tungnguyen/java/vn/asiantech/internship/CreateLogin/SignUpFragment.java [hunk 21]
```diff
@@ -1,4 +1,4 @@
-package vn.asiantech.internship;
+package vn.asiantech.internship.CreateLogin;
```
Comment (at-dev-support-01) [inline app/src/at-tungnguyen/java/vn/asiantech/internship/CreateLogin/SignUpFragment.java]: WARNING: [PackageCase] Package name contains upper case characters
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#PackageCase

**Comment by at-tungnguyen2** [comment]
@at-huongnguyen Please check for me, thank

File: app/src/at-tungnguyen/res/layout/item_status.xml [hunk 22]
```diff
@@ -0,0 +1,68 @@
...
+                android:textSize="20dp"
+                android:textStyle="bold"
+                tools:text="Thanh Tung" />
+
+            <ImageView
+                android:id="@+id/imgLike"
+                android:layout_width="50dp"
+                android:layout_height="50dp"
+                android:layout_marginLeft="5dp"
+                android:background="@drawable/selector_like_plus" />
```
Comment (at-huongnguyen) [inline app/src/at-tungnguyen/res/layout/item_status.xml]: Should rename to: selector_like

File: app/src/at-tungnguyen/res/layout/item_status.xml [hunk 23]
```diff
@@ -0,0 +1,68 @@
...
+
+            <ImageView
+                android:id="@+id/imgDislike"
+                android:layout_width="50dp"
+                android:layout_height="50dp"
+                android:layout_marginLeft="5dp"
+                android:background="@drawable/selector_dislike_sub" />
```
Comment (at-huongnguyen) [inline app/src/at-tungnguyen/res/layout/item_status.xml]: Should rename to: selector_dislike

File: app/src/at-tungnguyen/res/layout/item_status.xml [hunk 24]
```diff
@@ -0,0 +1,68 @@
+<?xml version="1.0" encoding="utf-8"?>
+<android.support.v7.widget.CardView xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="120dp"
```
Comment (at-huongnguyen) [inline app/src/at-tungnguyen/res/layout/item_status.xml]: Use wrap_content

File: app/src/at-tungnguyen/res/values/strings.xml [hunk 25]
```diff
@@ -19,6 +19,7 @@
     <string name="btn_division">/</string>
     <string name="tv_result">Result</string>
     <string name="tv_calculator">Calculator</string>
+    <string name="btn_ryclerview">RecyclerView</string>
```
Comment (at-huongnguyen) [inline app/src/at-tungnguyen/res/values/strings.xml]: btn_ryceclerview

**Comment by at-tungnguyen2** [comment]
@at-huongnguyen  Please check for me,thank you very much

**Pull request merged**